frontend/src/screens/Product/index.js

```diff
@@ -1,32 +1,83 @@
 import React, { Fragment, useEffect } from 'react';
 import useCustomContext from '../../customHooks/Hook';
 import axios from 'axios';
-import { PRODUCTS_LIST_SUCCESS, PRODUCTS_LIST_FAILURE, PRODUCTS_LIST_REQUEST } from '../../constants/ProductConstants';
 import { Loader, Sidebar, Message, Card, Searchbar } from '../../components';
-import { PRICE_HIGH_TO_LOW, PRICE_LOW_TO_HIGH } from '../../constants/FilterConstants';
+import Config from '../../config/Config';
+import { auth } from '../../utlis/auth';
+import { GET_CART_DATA, GET_WISHLIST_DATA, PRICE_HIGH_TO_LOW, PRICE_LOW_TO_HIGH, PRODUCTS_LIST_SUCCESS, PRODUCTS_LIST_FAILURE, PRODUCTS_LIST_REQUEST, FETCH_USER_DETAILS } from '../../constants/type';
+import { toast } from 'react-toastify';
+const { serverUrl } = Config
 
 const Products = () => {
 
     const { state, dispatch } = useCustomContext();
     const { loading, products, error } = state;
+    const token = auth();
 
     const fetchProducts = async () => {
         try {
             dispatch({ type: PRODUCTS_LIST_REQUEST })
-            const { data } = await axios.get('/api/products');
-            dispatch({ type: PRODUCTS_LIST_SUCCESS, payload: data.products })
+            const { data } = await axios.get(`${serverUrl}/api/products`, { headers: token });
+            dispatch({ type: PRODUCTS_LIST_SUCCESS, payload: data })
         } catch (err) {
             dispatch({ type: PRODUCTS_LIST_FAILURE, payload: 'Something went Wrong' })
+            const error = err.response.data.message;
+            toast.error(`${error}`);
+
         }
     }
```

Comment:
All the calls to server to fetch data can be moved into a separate file.

Reply:
I am not sure that whether that approach would be best to do because uptil now I have seen my seniors to keep server req in same file
Not sure of this. Need to discuss on more on this.
Thanks